www/components/Layout.js

```diff
@@ -13,19 +13,34 @@ const pagesWithoutDrawer = new Set([
   'register'
 ]);
 
+const pagesNeedingAuth = new Set([
+  'meeting',
+  'user',
+  '[id]'
+]);
+
+const selectUser = state => state.user;
+
 const Layout = ({ children }) => {
   const drawerOpen = useSelector( ( state ) => state.drawer );
   const storeTheme = useSelector( ( state ) => state.theme );
 
   const router = useRouter();
 
+  const user     = useSelector( selectUser );
+
   const page       = router.pathname.split('/').pop();
   const showDrawer = !pagesWithoutDrawer.has( page );
+  const blockPage = pagesNeedingAuth.has( page );
 
   useEffect( () => {
     document.documentElement.setAttribute(
       'data-theme', storeTheme.theme.theme
     );
+
+    if ( !user._id && blockPage ) {
+      router.push('/login');
```

Comment:
I was having that issue with the router taking a while to start. Does this work on a hard reload?

Reply:
@thudsonbu You said adding it as a dependency in the use effect will fix that right?

Reply:
Yeah, you can fix it by adding `router` as a dependency and only use the `router.push` if the router is fully initialized. You might want to experiment with how to check that the router is ready.

Reply:
I think there is a better solution for this but this should be good for now.